src/core/reducers/data/navigations.ts

```diff
@@ -26,11 +26,25 @@ export default function updateNavigations(state: State = DEFAULTS, action: Actio
 }
 
 export const updateSearch = (state: State, payload: Actions.Payload.Search) => {
-  if (payload.clear) {
-    const byId = state.allIds.reduce((navs, nav) =>
-      Object.assign(navs, { [nav]: { ...state.byId[nav], selected: [] } }), {});
+  if (payload.clear in state.byId) {
+    const navigationId = <string>payload.clear;
 
-    state = { ...state, byId };
+    state = {
+      ...state,
+      byId: {
+        ...state.byId,
+        [navigationId]: {
+          ...state.byId[navigationId],
+          selected: []
+        }
+      }
+    };
+  } else if (payload.clear) {
+    state = {
+      ...state,
+      byId: state.allIds.reduce((navs, nav) =>
+        Object.assign(navs, { [nav]: { ...state.byId[nav], selected: [] } }), {})
```

Comment:
Although I realize this was existing code, and this suggestion may be premature optimization, but I would do a direct assignment since it would be much more performant than doing an `Object.assign` for every nav.
```js
(navs, nav) => {
  navs[nav] = { ... state.byId[nav], selected: [] } };
  return navs;
}
```

Alternatively, as a one-liner:

```js
(navs, nav) => ((navs[nav] = { ... state.byId[nav], selected: [] } }) || nav)
```

I think the first form is clearer.